app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -665,8 +666,7 @@ public boolean onOptionsItemSelected(final MenuItem item) {
                 return true;
             case R.id.action_play_with_kodi:
                 try {
-                    NavigationHelper.playWithKore(activity, Uri.parse(
-                            url.replace("https", "http")));
+                    NavigationHelper.playWithKore(activity, Uri.parse(currentInfo.getUrl()));
```

Comment:
```suggestion
                    NavigationHelper.playWithKore(activity, Uri.parse(url));
```

Reply:
I changed this because url is the originalUrl, that can be in a non-standard form, while getUrl() gets the standard url

Reply:
Oh, you're right, I thought I had checked that…